nupic/research/frameworks/continuous_learning/multihead/subdataset.py

```diff
@@ -0,0 +1,191 @@
+                                   target_transform=None, download=download)
+    full_test_set = dataset_class("~/nta/continual-learning/datasets/mnist/",
+                                  train=False, transform=dataset_transform,
+                                  target_transform=None, download=download)
+
+    train_datasets, test_datasets = [], []
+
+    for labels in [[0, 1], [2, 3], [4, 5], [6, 7], [8, 9]]:
+
+        if scenario == "domain":
+            target_transform = transforms.Lambda(lambda y, x=labels[0]: y - x)
+        else:
+            target_transform = None
+
+        train_datasets.append(SubDataset(full_train_set, labels,
+                                         target_transform=target_transform))
+        test_datasets.append(SubDataset(full_test_set, labels,
+                                        target_transform=target_transform))
+
+    return train_datasets, test_datasets
+
+
+class GSC(Dataset):
+    """ Google Speech Commands dataset. """
+
+    def __init__(self, root=".", train=True, transform=None, target_transform=None,
+                 download=False):
+        super(GSC, self).__init__()
+
+        self.transform = transform
+        self.target_transform = target_transform
+
+        if train:
+            self.data = [np.load(os.path.join(
+                root, "gsc_train{}.npz".format(n)
+            ))["arr_0"] for n in range(30)]
+            self.data = np.concatenate(self.data)
+
+            self.targets = [np.load(os.path.join(
+                root, "gsc_train{}.npz".format(n)
+            ))["arr_1"] for n in range(30)]
+            self.targets = np.concatenate(self.targets)
+
+            # remove items with labels 0 or 1
+            idx = (self.targets > 1)
+            self.data = self.data[idx]
+            self.targets = self.targets[idx]
+
+            # shift all labels to be at 0
+            self.targets = self.targets - 2
+
+            # select subset of data
+            inds = np.random.choice(np.arange(len(self.data)), size=60000,
+                                    replace=False)
+            self.data = self.data[inds]
+            self.targets = self.targets[inds]
+
+        else:
+            self.data = np.concatenate((
+                np.load(os.path.join(root, "gsc_valid.npz"))["arr_0"],
+                np.load(os.path.join(root, "gsc_test_noise00.npz"))["arr_0"]
+            ))
+            self.test_labels = np.concatenate((
+                np.load(os.path.join(root, "gsc_valid.npz"))["arr_1"],
+                np.load(os.path.join(root, "gsc_test_noise00.npz"))["arr_1"]
+            ))
+
+            # remove items with labels 0 or 1
+            idx = (self.test_labels > 1)
+            self.data = self.data[idx]
+            self.test_labels = self.test_labels[idx]
+
+            # shift all labels to be at 0
+            self.test_labels = self.test_labels - 2
+
+            # select subset of data
```

Comment:
Best to remove commented code, if not relevant anymore

Reply:
Done.